sbatch_scripts/gen_sbatch.py

```diff
@@ -0,0 +1,406 @@
+        error (string): Name for the error file after running SBATCH job
+        time (string): Time limit for job
+        mem (string): Amount of memory in GB to be used with SBATCH job
+        command (string): Python command(s) to run after setting up all SBATCH metadata
+        environment (string): Additional metadata that includes commands to create the appropriate environment for generate or jaxtrain
+
+    Returns:
+        sbatch_script (sbatch_template): SBATCH script with metadata used for running SBATCH jobs
+
+    """
+
+    sbatch_script = SBATCH_TEMPLATE.format(
+        mem=mem,
+        job_name=job_name,
+        time=time,
+        output=output,
+        error=error,
+        command=command,
+        environment=environment
+    )
+    return sbatch_script
+
+def write_sbatch(script, sbatch_script):
+    """
+    Adds SBATCH metadata to .sh script
+
+    Parameters:
+        script (.sh file): .sh file that will be turned into an SBATCH script
+        sbatch_script (string): output from the create_sbatch_script function with SBATCH metadata
+
+    Returns:
+        None.
+    """
+    with open(script, "w") as f:
+        f.write(sbatch_script)
+
+def submit_sbatch(script, logger):
+    """
+    Submits SBATCH .sh file to OSCAR.
+
+    Parameters:
+        script (SBATCH .sh): script with SBATCH metadata added after running write_sbatch
+        logger (Logger): logger from file
+    """
+    try:
+        result = subprocess.run(["sbatch", script], capture_output=True, text=True)
+        logger.info(result.stdout)
+        if result.stderr:
+            logger.error(result.stderr)
+    except Exception as e:
+        logger.error(f"Failed to submit job: {e}")
+
+def get_basic_config_from_yaml(
+    yaml_config_path: str | Path
+):
+    """
+    Load the basic configuration from a YAML file. Modified from the generate.py file
+    
+    Parameters:
+        yaml_config_path (string or Path): path to the .yaml config file for use with generate.py
+
+    Returns:
+        basic_config (dictionary?): basic configuration dictionary, taken from the config.yaml file
+    
+    """
+    basic_config = yaml.safe_load(open(yaml_config_path, "rb"))
+    return basic_config
+
+def get_environment_setup(make_env: bool, command: str):
+    """
+    Sets up environment for running generate or jaxtrain if the --make-env argument is sued
+
+    Parameters:
+        make_env (Boolean): When True, sets up environment to run genreate or jaxtrain in the SBATCH script
+        command (str): Either "generate" or "jaxtrain" depending on which script you are running
+    """
+    if make_env:
+
+        if command == "generate":
+
+            environment = textwrap.dedent("""
+            module load python
+            module load gcc
+            git clone https://github.com/lnccbrown/ssm-simulators.git
+            cd ssm-simulators
+            git checkout add-generate-to-release-0.8.3
```

Comment:
This will clone the repo only if the repo hasn't been cloned already.
```suggestion
            if [ ! -d "ssm-simulators" ]; then
                git clone https://github.com/lnccbrown/ssm-simulators.git -b add-generate-to-release-0.8.3
            fi
            cd ssm-simulators
```
If you like you could make this more general?

Reply:
Wait why are we cloning `ssm-simulators` but using `uv add` with `LANfactory`? It's because we need the specific `add-generate-to-release-0.8.3` branch, right?

Reply:
I tried running generate with the environment setup
```
uv add ssm-simulators==0.9.0b5
uv sync
uv run ssms/cli/generate.py ...
```
and it gave me the error
```
error: Failed to spawn: `ssms/cli/generate.py`
  Caused by: No such file or directory (os error 2)
```
But when I ran generate with the environment
```
uv add ssm-simulators==0.9.0b5
cd ssm-simulators
uv sync
uv run ssms/cli/generate.py ...
```
it worked!

However, when I ran
```
uv add ssm-simulators==0.9.0b5
uv sync
uv run generate ...
```
I got the circular import error
```
Traceback (most recent call last):
  File "/users/hhafner/.local/bin/generate", line 5, in <module>
    from ssms.cli.generate import app
  File "/users/hhafner/.local/lib/python3.11/site-packages/ssms/__init__.py", line 2, in <module>
    from . import basic_simulators
ImportError: cannot import name 'basic_simulators' from partially initialized module 'ssms' (most likely due to a circular import) (/users/hhafner/.local/lib/python3.11/site-packages/ssms/__init__.py)
```
So I think having the `uv run` with the path to `generate.py` or `jax_train.py` is not the way it should be setup, it should be `uv run generate ...` so you won't need a `cd` command after adding `ssm-simulators` (when the circular import bug is fixed) or after adding `LANfactory`

Reply:
Right, long term it'd be nice to not have to clone the repo! But this workaround is okay for now :). Thanks for checking!